src/main/java/com/st/utopia/agent/controller/SeatTickets.java

```diff
@@ -0,0 +1,189 @@
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.client.RestTemplate;
+
+import com.st.utopia.agent.model.PaymentAmount;
+import com.st.utopia.agent.model.SeatLocation;
+import com.st.utopia.agent.model.Ticket;
+import com.st.utopia.agent.model.User;;
+
+/**
+ * Controller for agents using this user-facing interface
+ *
+ * @author Jonathan Lovelace
+ * @author Salem O
+ */
+@RestController
+public class SeatTickets {
+	@Autowired
+	private RestTemplate restTemplate;
+
+	@Value("${utopia.search.API}")
+	private String searchAPI;
+
+	@Value("${utopia.booking.API}")
+	private String bookingAPI;
+
+	@Value("${utopia.cancellation.API}")
+	private String cancellationAPI;
+	
+	/**
+	 * Helper method to reduce the amount of repetitive code required for "get-all"
+	 * methods.
+	 * @param url the URL to send the REST request to
+	 * @param <T> the type we expect
+	 * @return the response the server sent
+	 */
+	private <T> ResponseEntity<T> methodCall(final String url, final HttpMethod method) {
+		return restTemplate.exchange(url, method, null,
+				new ParameterizedTypeReference<T>() {});
+	}
+	
+	/**
+	 * Helper method to reduce the amount of repetitive code required for a method call.
+	 * @param url the URL to send the REST request to
+	 * @param type of method to send the REST request to
+	 * @param HttpEntity to send in the REST request
+	 * @param <T> the type we expect
+	 * @param <U> the type of the body sent to the server
+	 * @return the response the server sent
+	 */
+	private <T, U> ResponseEntity<T> methodCall(final String url, final HttpMethod method,
+			final HttpEntity<U> body) {
+		return restTemplate.exchange(url, method, body,
+				new ParameterizedTypeReference<T>() {});
+	}
+
+	/**
+	 * Get list of seats on a given flight
+	 * 
+	 * @return list of seats
+	 */
+	@GetMapping("/flight/{flightId}/seats")
+	public ResponseEntity<SeatLocation> getAllSeatsWithPlan(
+			@PathVariable final String flightId) {
+		// FIXME: Search service doesn't yet provide this
+		String url = searchAPI + "/seats?flight=" + flightId;
+		return this.<SeatLocation>methodCall(url, HttpMethod.GET);
+	}
+
+	/**
+	 * Get list of seats on a given flight
+	 * 
+	 * @return list of seats
+	 */
+	@GetMapping("/flight/{flightId}/seat/{row}/{seatId}")
+	public ResponseEntity<Ticket> getTicket(
+			@PathVariable final int flightId, @PathVariable final int row,
+			@PathVariable final String seatId) {
+		String url = bookingAPI + "/details/flights/" + flightId + "/rows/" +
```

Comment:
You might want to use `String.format` rather than repeated concatenation; it'll be shorter, and would be a little easier to debug in some cases if you forgot something.

Reply:
But, in the short time that we have, it would be difficult to do it for all of the concatenations.